include/bpf_helper_defs.h

```diff
@@ -309,3 +309,35 @@ EBPF_HELPER(uint64_t, bpf_get_current_pid_tgid, ());
 #ifndef __doxygen
 #define bpf_get_current_pid_tgid ((bpf_get_current_pid_tgid_t)BPF_FUNC_get_current_pid_tgid)
 #endif
+
+/**
+ * @brief Get the 64-bit logon ID of the current process. In case of sock_addr
+ * attach types, get the logon ID of the user mode app making the request. In other
+ * cases, get the logon ID of the current thread.
+ *
+ * @param[in] ctx Context passed to the eBPF program.
+ * @param[out] logon_id On success, contains the logon ID value.
+ * @param[in] size Size of uint64_t.
+ *
+ * @returns 0 if the operation was successful, 1 otherwise.
+ */
+EBPF_HELPER(uint32_t, bpf_get_current_logon_id, (void* ctx, uint64_t* logon_id, int size));
+#ifndef __doxygen
+#define bpf_get_current_logon_id ((bpf_get_current_logon_id_t)BPF_FUNC_get_current_logon_id)
+#endif
+
+/**
+ * @brief Get whether the current user is admin. In case of sock_addr attach types,
+ * returns whether the user initiating the request is admin or not. In other
+ * cases, returns whether the current thread user is admin or not.
+ *
+ * @param[in] ctx Context passed to the eBPF program.
+ * @param[out] is_admin Value whether user is admin or not.
+ * @param[in] size Size of uint32_t.
```

Comment:
This looks weird! I am pretty sure this is to pass the verifier - but looks odd. Can we maybe make the function return uint32_t instead and returns 0 on error?

Reply:
Also if the return value is supposed to be error , shouldnt it return -ve integer for error; for instance `bpf_map_update_elem`

Reply:
I looked at existing Linux helper functions to get idea how to return error code. One example is `bpf_fib_lookup`. 

```
long bpf_fib_lookup(void *ctx, struct bpf_fib_lookup *params, int
       plen, u32 flags)
```

`bpf_fib_lookup` returns the following error codes:

```
Return

                     • < 0 if any input argument is invalid

                     • 0 on success (packet is forwarded, nexthop
                       neighbor exists)

                     • > 0 one of BPF_FIB_LKUP_RET_ codes explaining why
                       the packet is not forwarded or needs assist from
                       full stack
```

Reply:
Also we need to differentiate between 3 scenarios:
1. is admin
2. is not admin
3. failed to detemine

Reply:
<0 failed to determine
0 is not admin
1 is admin

Reply:
Yes we can do that. I think then we can do the same for `bpf_get_current_logon_id` also:

<0 error 
>=0 valid logon id

Reply:
I will make these changes.

Reply:
Updated the helper functions.